compiler/lib/src/main/scala/codegen/CppWriter/PortCppWriter.scala

```diff
@@ -0,0 +1,624 @@
+      n,
+      paramAnnotationMap(n)
+    )
+  })
+
+  // Return type as a C++ type
+  private val returnType = data.returnType match {
+    case Some(value) => s.a.typeMap(value.id) match {
+      case t: Type.String => strCppWriter.getQualifiedClassName(t, List(strNamespace))
+      case t => typeCppWriter.write(t)
+    }
+    case None => "void"
+  }
+
+  // Translate formal parameter to C++ parameter
+  private def getCppType(t: Type, typeName: String, kind: Ast.FormalParam.Kind) =
+    (t, kind)  match {
+      // Reference formal parameters become non-constant C++ reference parameters
+      case (_, Ast.FormalParam.Ref) => s"$typeName&"
+      // Primitive, non-reference formal parameters become C++ value parameters
+      case (t, Ast.FormalParam.Value) if s.isPrimitive(t, typeName) => typeName
+      // Other non-reference formal parameters become constant C++ reference parameters
+      case (_, Ast.FormalParam.Value) => s"const $typeName&"
+    }
+
+  private def writeIncludeDirectives: List[String] = {
+    val Right(a) = UsedSymbols.defPortAnnotatedNode(s.a, aNode)
+    s.writeIncludeDirectives(a.usedSymbolSet)
+  }
+
+  def write: CppDoc = {
+    val includeGuard = s.includeGuardFromQualifiedName(symbol, fileName)
+    CppWriter.createCppDoc(
+      s"$name port",
+      fileName,
+      includeGuard,
+      getMembers,
+      s.toolName
+    )
+  }
+
+  private def getMembers: List[CppDoc.Member] = {
+    val hppIncludes = getHppIncludes
+    val cppIncludes = getCppIncludes
+    val annotation = AnnotationCppWriter.asStringOpt(aNode) match {
+      case Some(value) => s"\n$value"
+      case None => ""
+    }
+    val portBufferClass = data.returnType match {
+      case Some(_) => Nil
+      case None => List(
+        CppDoc.Member.Lines(
+          CppDoc.Lines(
+            Line.blank :: wrapInAnonymousNamespace(getPortBufferClass),
+            CppDoc.Lines.Cpp
+          )
+        )
+      )
+    }
+    val classes = List(
+      getStringClasses,
+      getStringTypedefs,
+      portBufferClass,
+      List(
+        CppDoc.Member.Class(
+          CppDoc.Class(
+            Some(s"Input $name port" + annotation),
+            s"Input${name}Port",
+            Some("public Fw::InputPortBase"),
+            getInputPortClassMembers
+          )
+        ),
+        CppDoc.Member.Class(
+          CppDoc.Class(
+            Some(s"Output $name port" + annotation),
+            s"Output${name}Port",
+            Some("public Fw::OutputPortBase"),
+            getOutputPortClassMembers
+          )
+        ),
+      )
+    ).flatten
+    List(
+      List(hppIncludes, cppIncludes),
+      CppWriter.wrapInNamespaces(namespaceIdentList, classes)
+    ).flatten
+  }
+
+  private def getHppIncludes: CppDoc.Member = {
+    val serializableHeader = data.returnType match {
+      case Some(_) => Nil
+      case None => List("Fw/Types/Serializable.hpp")
+    }
+    val standardHeaders = (
+      List(
+        "cstdio",
```

Comment:
cstdio and cstring should be system headers <...> not standard headers "...".

Reply:
Compare how we handle system headers in the other Cpp writers.

Reply:
Fixed!